### PR TITLE
[kuberay][docs] Experimental features

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
Signed-off-by: Dmitri Gekhtman <dmitri.m.gekhtman@gmail.com>

## Why are these changes needed?

This doc gives an overview of experimental KubeRay features, with links to the KubeRay docs for more detailed info.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
cc @edoakes 
There might be an overlap with serve docs.

File: doc/source/cluster/kubernetes/user-guides/experimental.md
Comment (edoakes) [inline doc/source/cluster/kubernetes/user-guides/experimental.md:16]: as we discussed please update link to point at the serve k8s user guide once this is merged:
https://github.com/ray-project/ray/pull/27747

**Comment by DmitriGekhtman** [comment]
This needs an edit after https://github.com/ray-project/ray/pull/27747 is merged.
Will hold off on this PR until that one is in.

**Comment by DmitriGekhtman** [comment]
@edoakes I've added the requested internal links, ptal

File: doc/source/cluster/kubernetes/user-guides/experimental.md
Comment (DmitriGekhtman) [reply doc/source/cluster/kubernetes/user-guides/experimental.md]: Fixed!

**Review by edoakes — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
